ocaml/xapi/repository.ml

```diff
@@ -720,40 +668,41 @@ let apply_updates' ~__context ~host ~updates_info ~livepatches ~acc_rpm_updates
             )
         )
       ]
-      ) ;
-  (* Evaluate guidances *)
-  let guidances =
-    let guidances' =
-      (* EvacuateHost will be applied before applying updates *)
-      eval_guidances ~updates_info ~updates:acc_rpm_updates ~kind:Recommended
-        ~livepatches:successful_livepatches ~failed_livepatches
-      |> List.filter (fun g -> g <> Guidance.EvacuateHost)
-      |> fun l -> merge_with_unapplied_guidances ~__context ~host ~guidances:l
-    in
-    GuidanceSet.assert_valid_guidances guidances' ;
-    match failed_livepatches with
-    | [] ->
-        guidances'
-    | _ :: _ ->
-        (* There is(are) livepatch failure(s):
-         * the host should not be rebooted, and
-         * an extra pending guidance 'RebootHostOnLivePatchFailure' should be set.
-         *)
-        guidances'
-        |> List.filter (fun g -> g <> Guidance.RebootHost)
-        |> List.cons Guidance.RebootHostOnLivePatchFailure
+      )
+
+let apply_updates' ~__context ~host ~updates_info ~livepatches ~acc_rpm_updates
+    =
+  let open Guidance in
+  (* This function runs on coordinator host *)
+  let get_guidances kind =
+    eval_guidances ~updates_info ~updates:acc_rpm_updates ~kind ~livepatches
+    |> GuidanceSet.remove EvacuateHost
+    |> GuidanceSet.elements
   in
+  let mandatory = get_guidances Mandatory in
+  (* Install RPM updates firstly *)
+  Helpers.call_api_functions ~__context (fun rpc session_id ->
+      Client.Client.Repository.apply ~rpc ~session_id ~host
+  ) ;
```

Comment:
As it is the first thing done in this function, would be clear to move these lines to the start of this function as original.

Reply:
Updated the comment. This is not the first thing :)